crypto/fipsmodule/rand/rand.c

```diff
@@ -489,18 +496,24 @@ static void rand_bytes_private(uint8_t *out, size_t out_len,
 
     rand_state_initialize(state);
     thread_local_list_add_node(state);
+
+    ctr_drbg_state_is_fresh = 1;
   }
 
   rand_bytes_core(state, out, out_len, user_pred_resistance,
-    use_user_pred_resistance);
+    use_user_pred_resistance, ctr_drbg_state_is_fresh);
 
   FIPS_service_indicator_unlock_state();
   FIPS_service_indicator_update_state();
 }
 
 int RAND_bytes_with_user_prediction_resistance(uint8_t *out, size_t out_len,
   const uint8_t user_pred_resistance[RAND_PRED_RESISTANCE_LEN]) {
-  
+
+  if (user_pred_resistance == NULL) {
+    abort();
+  }
```

Comment:
Why not return 0 for error? I also checked and rand_bytes_core uses GUARD_PTR_ABORT so this should at least be consistent. But I'm still interested why all the aborts on invalid inputs.

Reply:
I was debating that. Could return 0. But then I changed to abort because wanted to stay consistent with the fact that all failure modes result in an abort.

Reply:
I think I lean towards following the current "abort" pattern.